augur/utils.py

```diff
@@ -17,6 +19,31 @@
 class AugurException(Exception):
     pass
 
+@contextmanager
+def open_file(fname, mode):
+    """Open a file using either gzip.open() or open() depending on file name. Semantics identical to open()"""
+    if fname.endswith('.gz'):
+        if "t" not in mode:
+            # For interoperability, gzip needs to open files in "text" mode
+            mode = mode + "t"
+        with gzip.open(fname, mode) as fh:
+            yield fh
+    else:
+        with open(fname, mode) as fh:
+            yield fh
+
+def is_vcf(fname):
```

Comment:
Several contributors have started to add their own implementations of this function (or similar functions). We should doublecheck that this function isn't redundant with an existing function when we finally merge this PR.

Reply:
Yeah, I think utils.py is generally underutilized right now. I'm fine with whatever the canonical implementation is as long as there IS a canonical implementation